### PR TITLE
TF-3023 SMime signature validation

### PR DESCRIPTION
**Pull request by dab246**
## Issue

#3023
#3031

## Resolved

- Good signature:

https://github.com/user-attachments/assets/beef0b8b-819e-4e4d-9ddd-54f562664349

- Bad signature:

https://github.com/user-attachments/assets/8f8c216e-3094-4c29-9317-135381a5f1ed

### CODE REVIEW

File: test/features/email/presentation/information_sender_and_receiver_builder_widget_test.dart [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+        await tester.pumpWidget(widget);
+
+        await tester.pumpAndSettle();
+
+        expect(
+          find.byKey(const Key('smime_signature_status_icon')),
+          findsOneWidget);
+      });
+
+      testWidgets(
```
Comment (hoangdat) [inline test/features/email/presentation/information_sender_and_receiver_builder_widget_test.dart:65]: please also test with `Good Signatures` case

File: test/features/email/presentation/information_sender_and_receiver_builder_widget_test.dart [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+          findsOneWidget);
+
+        final sMimeSignatureStatusIconWidgetFinder = find.byKey(const Key('smime_signature_status_icon'));
+        final sMimeSignatureStatusIconWidget = tester.widget<Tooltip>(sMimeSignatureStatusIconWidgetFinder);
+        expect(
+          sMimeSignatureStatusIconWidget.message,
+          'The authenticity of this message had been verified with SMime signature.');
+      });
+
+      testWidgets(
```
Comment (hoangdat) [inline test/features/email/presentation/information_sender_and_receiver_builder_widget_test.dart:102]: test with `Bad Signatures`

File: lib/features/email/presentation/extensions/presentation_email_extension.dart [hunk 3]
```diff
@@ -0,0 +1,18 @@
+
+import 'package:model/email/presentation_email.dart';
+import 'package:model/extensions/list_email_header_extension.dart';
+import 'package:tmail_ui_user/features/email/presentation/model/smime_signature_status.dart';
+
+extension PresentationEmailExtension on PresentationEmail {
+
+  SMimeSignatureStatus get sMimeStatus {
+    final status = emailHeader?.toSet().sMimeStatus;
+    if (status == 'Good signature') {
```
Comment (hoangdat) [inline lib/features/email/presentation/extensions/presentation_email_extension.dart:10]: please define constant for 'Good signature' and 'Bad signature' in SMimeConstant

Comment (dab246) [reply lib/features/email/presentation/extensions/presentation_email_extension.dart]: done

File: test/features/email/presentation/information_sender_and_receiver_builder_widget_test.dart (hunk 2 above)
Comment (dab246) [reply test/features/email/presentation/information_sender_and_receiver_builder_widget_test.dart:102]: done

File: test/features/email/presentation/information_sender_and_receiver_builder_widget_test.dart (hunk 1 above)
Comment (dab246) [reply test/features/email/presentation/information_sender_and_receiver_builder_widget_test.dart:65]: done

**Comment by chibenwa** [comment]
Please add handling for `X-SMIME-Status: Unknown` (yellow with question mark + explanation: This message was signed with an unknown authority: we are unable to verify this signature)

CF https://github.com/apache/james-project/pull/2366#pullrequestreview-2220202747

**Review by hoangdat — APPROVED** [review]

**Comment by dab246** [comment]
- [ ] Rebase

**Comment by hoangdat** [comment]
please rebase

**Comment by dab246** [comment]
> please rebase

Done

**Review by tddang-linagora — APPROVED** [review]

**Comment by hoangdat** [comment]
- [ ] rebase with refactor branch
- [ ] test with real EML from MU

**Comment by dab246** [comment]
> * [x]  rebase with refactor branch
> * [ ]  test with real EML from MU

@hoangdat  `eml` file from MU with`Email/get` does not return header containing `X-SMIME-Status`

**Comment by Arsnael** [comment]
> @hoangdat `eml` file from MU with`Email/get` does not return header containing `X-SMIME-Status`

That's likely because you need to have the SmimeCheckSignature mailet added in the mailetcontainer.xml file for tmail-backend with the cert from MU, so that when the backend is processing the eml file example you got from MU, add the header X-SMIME-Status which you can read

**Comment by hoangdat** [comment]
hi @Arsnael , v0.13.0 was release in `canary-tmail (linagora)` or `mail.stg.lin-saas.com (TWP)`. Can you take a look and test for SMime? Thanks

**Pull request merged**